nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+        elif isinstance(item1, list) and isinstance(item2, list):
+            for our_item in item1:
+                if not self._recursive_check_item_against_list(our_item, item2):
+                    return False
+        else:
+            if isinstance(item1, str) and item1.isdigit():
+                item1 = int(item1)
+            if isinstance(item2, str) and item2.isdigit():
+                item2 = int(item2)
+            if item1 != item2:
+                return False
+        return True
+
+    def _recursive_check_item_against_list(self, item1, list2: list) -> bool:
+        if isinstance(item1, list):
+            raise NotImplementedError
+        elif isinstance(item1, dict):
+            for item2 in list2:
+                if isinstance(item2, list):
+                    raise NotImplementedError
+                elif isinstance(item2, dict):
+                    if self._recursive_item_check(item1, item2):
+                        return True
+                else:
+                    raise NotImplementedError
+        else:
+            raise NotImplementedError
+        return False
+
+    def subcommand(self, **kwargs):
+        """Makes a function into a subcommand."""
+        if self.type != ApplicationCommandType.chat_input:  # At this time, non-slash commands cannot have Subcommands.
+            raise TypeError(f"{self.type} cannot have subcommands.")
+        else:
+            def decorator(func: Callable):
+                result = ApplicationSubcommand(func, self, ApplicationCommandOptionType.sub_command, **kwargs)
+                self.children[result.name] = result
+                return result
+
+            return decorator
+
+
+def slash_command(*args, **kwargs):
+    """Creates a Slash Application Command, used inside of a ClientCog."""
+    def decorator(func: Callable):
+        if isinstance(func, ApplicationCommand):
+            raise TypeError("Callback is already an ApplicationCommandRequest.")
+        return ApplicationCommand(func, cmd_type=ApplicationCommandType.chat_input, *args, **kwargs)
+    return decorator
+
+
+def message_command(*args, **kwargs):
+    """Creates a Message Application Command, used inside of a ClientCog."""
+    def decorator(func: Callable):
+        if isinstance(func, ApplicationCommand):
+            raise TypeError("Callback is already an ApplicationCommandRequest.")
+        return ApplicationCommand(func, cmd_type=ApplicationCommandType.message, *args, **kwargs)
+    return decorator
+
+
+def user_command(*args, **kwargs):
+    """Creates a User Application Command, used inside of a ClientCog."""
+    def decorator(func: Callable):
+        if isinstance(func, ApplicationCommand):
+            raise TypeError("Callback is already an ApplicationCommandRequest.")
+        return ApplicationCommand(func, cmd_type=ApplicationCommandType.user, *args, **kwargs)
+    return decorator
+
+
+def options_to_args(options: List[Dict[str, Any]]) -> Dict[str, Any]:
```

Comment:
Given the usage, shouldn't ``str`` instead be typed as ``Literal["name", "value']``? Given that is all that's accessed or can other things be there, just ignored

Reply:
I don't even use this anymore?? Thanks for making me look at it, lol.